encfs/Context.cpp

```diff
@@ -46,9 +49,14 @@ EncFS_Context::~EncFS_Context() {
   openFiles.clear();
 }
 std::shared_ptr<DirNode> EncFS_Context::getRoot(int *errCode) {
-  std::shared_ptr<DirNode> ret;
+  std::shared_ptr<DirNode> ret = nullptr;
   do {
     {
+      Lock locku(unmountMutex, false);
+      if(!locku.isLocked()) {
```

Comment:
Mmmmh I think there might be an issue here if 2 IO operations call `getRoot()` at the same time, one of the 2 may be dropped...
This non-bloking lock allows `unmount()` to proceed correctly, but I finally think there is a drawback...

Reply:
Other solution found, I'll submit it